modules/formulize/admin/save/element_options_save.php

```diff
@@ -232,23 +233,30 @@
   }
 }
 
-		// check to see if we should be reassigning user submitted values, and if so, trap the old ele_value settings, and the new ones, and then pass off the job to the handling function that does that change
-		if(isset($_POST['changeuservalues']) AND $_POST['changeuservalues']==1) {
-      include_once XOOPS_ROOT_PATH . "/modules/formulize/class/data.php";
-			$data_handler = new formulizeDataHandler($fid);
-			switch($ele_type) {
-				case "radio":
-				case "check":
-					$newValues = $processedValues['elements']['ele_value'];
-					break;
-				case "select":
-					$newValues = $processedValues['elements']['ele_value'][2];
-					break;
-			}
-			if(!$changeResult = $data_handler->changeUserSubmittedValues($ele_id, $newValues)) {
-				print "Error updating user submitted values for the options in element $ele_id";
-			}
-		}
+// check to see if we should be reassigning user submitted values, and if so, trap the old ele_value settings, and the new ones, and then pass off the job to the handling function that does that change
+if(isset($_POST['changeuservalues']) AND $_POST['changeuservalues']==1) {
+  include_once XOOPS_ROOT_PATH . "/modules/formulize/class/data.php";
+  $data_handler = new formulizeDataHandler($fid);
+  switch($ele_type) {
+    case "radio":
+    case "check":
+      $newValues = $processedValues['elements']['ele_value'];
+      break;
+    case "select":
+      $newValues = $processedValues['elements']['ele_value'][2];
+      break;
+  }
+  if(!$changeResult = $data_handler->changeUserSubmittedValues($ele_id, $newValues)) {
+    print "Error updating user submitted values for the options in element $ele_id";
+  }
+}
+
+//newly added for autocomplete box to make sure when {USERNAMES} and {FULLNAMES} are selected, system will not allow new entries to be added
+//ele_value[8] ==1 will make sure it's an autocomplete box
+//Added by Jinfu MAR 2015
+if($processedValues['elements']['ele_value'][8]==1&&($processedValues['elements']['ele_value'][2]['{USERNAMES}']==1||$processedValues['elements']['ele_value'][2]['{FULLNAMES}']==1)){

```

Comment:
This line is hard to read because it is long and there are no spaces around the == and || operators. Also, it didn't work until I changed ['{USERNAMES}']==1 to ['{USERNAMES}']==0. Does it work with 1 for you?